lib/ansible/plugins/connection/psrp.py

```diff
@@ -479,11 +479,17 @@ def exec_command(self, cmd: str, in_data: bytes | None = None, sudoable: bool =
             pwsh_in_data = in_data
             display.vvv(u"PSRP: EXEC %s" % script, host=self._psrp_host)
 
-        rc, stdout, stderr = self._exec_psrp_script(
-            script=script,
-            input_data=pwsh_in_data.splitlines() if pwsh_in_data else None,
-            arguments=script_args,
-        )
+        try:
+            rc, stdout, stderr = self._exec_psrp_script(
+                script=script,
+                input_data=pwsh_in_data.splitlines() if pwsh_in_data else None,
+                arguments=script_args,
+            )
+        except ReadTimeout as e:
+            raise AnsibleConnectionFailure(
+                "Failed to read from the host via PSRP: %s"
+                % to_native(e)
+            )
```

Comment:
```suggestion
            raise AnsibleConnectionFailure(
                "HTTP read timeout during PSRP script execution"
            ) from e
```

Since Ansible 2.19 we try and wrap the exception in a more Pythonic manner with `raise Exception("msg") from e`. In this case the exception message will still contain the gist of the failure which we know is a HTTP read timeout but the stacktrace will still be present for the underlying exception if a user opts into viewing them.